nexus/db-queries/src/db/datastore/disk.rs

```diff
@@ -652,4 +655,143 @@ impl DataStore {
             }
         }
     }
+
+    /// Set a disk to faulted and un-delete it
+    ///
+    /// If the disk delete saga unwinds, then the disk should _not_ remain
+    /// deleted: disk delete saga should be triggered again in order to fully
+    /// complete, and the only way to do that is to un-delete the disk. Set it
+    /// to faulted to ensure that it won't be used.
+    pub async fn project_undelete_disk_set_faulted_no_auth(
+        &self,
+        disk_id: &Uuid,
+    ) -> Result<(), Error> {
+        use db::schema::disk::dsl;
+        let conn = self.pool_connection_unauthorized().await?;
+
+        let faulted = api::external::DiskState::Faulted.label();
+
+        let result = diesel::update(dsl::disk)
+            .filter(dsl::time_deleted.is_not_null())
+            .filter(dsl::id.eq(*disk_id))
```

Comment:
This is the primary key, right? I think `diesel::update(dsl::disk).find(*disk_id)` might be more clear here, unless I'm missing something about the DSL that precludes that.

Reply:
Ah, I think you might have included the filter on `time_deleted IS NOT NULL` to ensure you're only operating on deleted disks, for the benefit of the `execute_and_check()` mechanism. Ignore the previous suggestion if that's the case!

Reply:
:+1: